static/app/components/searchQueryBuilder/tokenizedQueryGrid.tsx

```diff
@@ -40,14 +43,35 @@ function useApplyFocusOverride(state: ListState<ParseResultToken>) {
 
 function Grid(props: GridProps) {
   const ref = useRef<HTMLDivElement>(null);
+  const selectionKeyHandlerRef = useRef<HTMLInputElement>(null);
   const {size} = useSearchQueryBuilder();
-  const {state, gridProps} = useQueryBuilderGrid(props, ref);
-
+  const state = useListState<ParseResultToken>({
```

Comment:
Had to move some things around to pass the `undo` function into the new component.